### PR TITLE
feat(liquidator & disputer): Further bot optimization

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

There are a few places in the liquidator and dispute bots that can be parallelized even further. This PR addresses this.

**Summary**

The update logic calls and gas estimation calls in `liquidator.js` and `disputer.js` were replaced with `Promise.all` syntax.

### CODE REVIEW

File: disputer/disputer.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ class Disputer {
 
       const txnConfig = {
         from: this.account,
-        gas: Math.min(gasLimit, this.txnGasLimit),
+        gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
```
Comment (nicholaspai) [inline disputer/disputer.js:151]: Should we set `cost gasToSend = Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit)` right after we calculate the `gasEstimation`? Since we re-use this calculation twice

Comment (nicholaspai) [reply disputer/disputer.js:151]: AH nevermind, I see the `gasEstimations` are for different txns

File: liquidator/liquidator.js [hunk 2]
```diff
@@ -412,11 +417,9 @@ class Liquidator {
       let resolvedPrice;
       if (requestTimestamp) {
         try {
-          resolvedPrice = revertWrapper(
-            await this.votingContract.getPrice(this.empIdentifier, requestTimestamp, {
-              from: this.empContract.options.address
-            })
-          );
+          resolvedPrice = await this.votingContract.getPrice(this.empIdentifier, requestTimestamp, {
```
Comment (nicholaspai) [inline liquidator/liquidator.js:420]: Why did you remove the `revertWrapper`? Sometimes `getPrice` will return a really weird number instead of throwing an error which is why we catch that

File: disputer/disputer.js [hunk 3]
```diff
@@ -225,10 +225,12 @@ class Disputer {
       const withdraw = this.empContract.methods.withdrawLiquidation(liquidation.id, liquidation.sponsor);
 
       // Confirm that dispute has eligible rewards to be withdrawn.
-      let withdrawAmount, gasLimit;
+      let withdrawAmount, gasEstimation;
       try {
-        withdrawAmount = revertWrapper(await withdraw.call({ from: this.account }));
```
Comment (nicholaspai) [inline disputer/disputer.js:230]: I don't think you want to remove the `revertWrapper`, just call `revertWraper` on the return value of `.call()`

File: liquidator/liquidator.js [hunk 4]
```diff
@@ -359,10 +361,13 @@ class Liquidator {
       const withdraw = this.empContract.methods.withdrawLiquidation(liquidation.id, liquidation.sponsor);
 
       // Confirm that liquidation has eligible rewards to be withdrawn.
-      let withdrawAmount, gasLimit;
+      let withdrawAmount, gasEstimation;
       try {
-        withdrawAmount = revertWrapper(await withdraw.call({ from: this.account }));
```
Comment (nicholaspai) [inline liquidator/liquidator.js:364]: Same thing here about `revertWrapper`

File: liquidator/liquidator.js (hunk 2 above)
Comment (chrismaree) [reply liquidator/liquidator.js]: ye, this is a good point. I've added it back accordingly as well as a comment with reference to the OZ thread on why this is needed.

**Comment by chrismaree** [comment]
@nicholaspai I've added back the revert wrappers. LMK what you think.
